apps/desktop/src/components/v3/IntegrateUpstreamModal.svelte

```diff
@@ -170,23 +163,20 @@
 	}
 
 	async function fetchAppliedBranches() {
-		return await stackService
-			.fetchStacks(projectId)
-			.then(
-				(stacksResponse) =>
-					stacksResponse.data?.flatMap((stack) => stack.heads.map((head) => head.name)) ?? []
-			);
+		const stacksResponse = await stackService.fetchStacks(projectId);
+		return stacksResponse.data?.flatMap((stack) => stack.heads.map((head) => head.name)) ?? [];
 	}
 
 	export async function show() {
 		integratingUpstream = 'inert';
 		branchStatuses = undefined;
-		filteredReviews = undefined;
+		filteredReviews = [];
```

Comment:
When we start to show the modal, clear out the old value of `filteredReviews` with an empty array. We are going to set it "synchronously" before `branchStatuses` (the thing the loading state depends on) gets set.